frontend/src/metabase/public/containers/PublicApp/PublicApp.unit.spec.tsx

```diff
@@ -32,7 +34,15 @@ function setup({
         <h1 data-testid="test-content">Test</h1>
       </EmbedFrame>
     </PublicApp>,
-    { mode: "public", storeInitialState: { app, settings }, withRouter: true },
+    {
+      mode: "public",
+      storeInitialState: { app, settings },
+      router: {
+        currentRoute: "/public/dashboard/:id",
+        currentLocation: `/public/dashboard/UUID${hash}`,
+      },
```

Comment:
Open questions about this change:

1. How clear is the difference between `route` and `location` here?
2. Does the `withRouter` option make any sense now? Should we maybe always expect the `router` object passed when we need a component wrapped in `react-router`?

Reply:
@kulyk btw is is **current** route? Based on the implementation it seems to be the initial one

Reply:
Good questions. Doesn't the _initial_ route become _current_ with our setup? 🤔 We only render a single route

Reply:
maybe not. i.e. `currentLocation` can be changed

Reply:
Yeah, that's possible, but it shouldn't come to this IMO

Reply:
I think it would be great to allow changing routes within a test

Reply:
If it doesn't work already, that's because of how `renderWithProviders` is implemented

Reply:
Sure, I was saying this mostly to think about possible uses cases for this new parameter. We can always change it later. The PR is approved!